### PR TITLE
JAMES-2586 Refactor/Cleanup JamesPostgresConnectionFactory

### PR DESCRIPTION
**Pull request by vttranlina**
When I tried to refactor PostgresExtension (Using PoolBackedPostgresConnectionFactory for all factories. Make it similar to prod code ([PostgresCommonModule](https://github.com/apache/james-project/blob/cfeba0162d4fa7c8549069649976f2cac45b628b/server/container/guice/postgres-common/src/main/java/org/apache/james/modules/data/PostgresCommonModule.java) ) )

I realized that `PoolBackedPostgresConnectionFactory` doesn't set attribute `app.current_domain` when Domain is empty. 
=> Failed on some test cases with enableRLS +  emptyDomain username

### How to fix it?
 set domain attribute value even when empty domain, make it similar to that we did in [DomainImplPostgresConnectionFactory](https://github.com/apache/james-project/blob/701790ec040011269640e13e416c5283ad649322/backends-common/postgres/src/main/java/org/apache/james/backends/postgres/utils/DomainImplPostgresConnectionFactory.java#L37)

### CODE REVIEW

**Comment by vttranlina** [comment]
After discussing with @hungphan227 @quantranhong1999 
we should not try to set an empty value for `app.current_domain` when input is empty Domain

-  About test cases failed, If the reason is `enable RLS` + `User has empty domain` + `test cases focus on business logic (not RLS logic) ` => change `PostgresExtension.withRowLevelSecurity` => `PostgresExtension.withoutRowLevelSecurity`

File: backends-common/postgres/src/main/java/org/apache/james/backends/postgres/utils/SinglePostgresConnectionFactory.java
Comment (vttranlina) [inline backends-common/postgres/src/main/java/org/apache/james/backends/postgres/utils/SinglePostgresConnectionFactory.java:1]: it is no longer worth using

File: backends-common/postgres/src/main/java/org/apache/james/backends/postgres/utils/DomainImplPostgresConnectionFactory.java
Comment (vttranlina) [inline backends-common/postgres/src/main/java/org/apache/james/backends/postgres/utils/DomainImplPostgresConnectionFactory.java:1]: The dicussion about removing it at https://github.com/apache/james-project/pull/2190#discussion_r1571991600

**Comment by vttranlina** [comment]
rebase `postgresql`

**Review by hungphan227 — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**